integrations/amazon_bedrock/tests/test_chat_generator.py

```diff
@@ -447,12 +454,162 @@ def test_callback(chunk: StreamingChunk):
+        """
+        Integration test that the AmazonBedrockChatGenerator component can run with the Haystack tools parameter.
+        and the streaming_callback parameter to get the streaming response.
+        """
+        initial_messages = [ChatMessage.from_user("What's the weather like in Paris?")]
+        component = AmazonBedrockChatGenerator(model=model_name, tools=tools, streaming_callback=print_streaming_chunk)
+        results = component.run(messages=initial_messages)
+
+        assert len(results["replies"]) > 0, "No replies received"
+
+        # Find the message with tool calls
+        tool_message = None
+        for message in results["replies"]:
+            if message.tool_call:
+                tool_message = message
+                break
+
+        assert tool_message is not None, "No message with tool call found"
+        assert isinstance(tool_message, ChatMessage), "Tool message is not a ChatMessage instance"
+        assert ChatMessage.is_from(tool_message, ChatRole.ASSISTANT), "Tool message is not from the assistant"
+
+        tool_call = tool_message.tool_call
+        assert tool_call.id, "Tool call does not contain value for 'id' key"
+        assert tool_call.tool_name == "weather"
+        assert tool_call.arguments == {"city": "Paris"}
+        assert tool_message.meta["finish_reason"] == "tool_use"
+
+        new_messages = [
+            initial_messages[0],
+            tool_message,
+            ChatMessage.from_tool(tool_result="22° C", origin=tool_call),
+        ]
+        # Pass the tool result to the model to get the final response
+        results = component.run(new_messages)
+
+        assert len(results["replies"]) == 1
+        final_message = results["replies"][0]
+        assert not final_message.tool_call
+        assert len(final_message.text) > 0
+        assert "paris" in final_message.text.lower()
+
+    @pytest.mark.parametrize("model_name", [MODELS_TO_TEST_WITH_TOOLS[0]])  # just one model is enough
+    @pytest.mark.integration
+    def test_pipeline_with_amazon_bedrock_chat_generator(self, model_name, tools):
+        """
+        Test that the AmazonBedrockChatGenerator component can be used in a pipeline
+        """
+
+        pipeline = Pipeline()
+        pipeline.add_component("generator", AmazonBedrockChatGenerator(model=model_name, tools=tools))
+        pipeline.add_component("tool_invoker", ToolInvoker(tools=tools))
+
+        pipeline.connect("generator", "tool_invoker")
+
+        results = pipeline.run(
+            data={"generator": {"messages": [ChatMessage.from_user("What's the weather like in Paris?")]}}
+        )
+
+        assert (
+            "The weather in Paris is sunny and 32°C"
+            == results["tool_invoker"]["tool_messages"][0].tool_call_result.result
+        )
+
+    @pytest.mark.parametrize("model_name", [MODELS_TO_TEST_WITH_TOOLS[0]])  # just one model is enough
+    @pytest.mark.integration
```

Comment:
```suggestion
```

- let's transform this into a unit test (with mocking?)
- if possible, let's test the dictionary format more extensively
- let's also dump to YAML and reload from YAML (some serialization errors happen in this phase)

An example of what I mean: https://github.com/deepset-ai/haystack-core-integrations/blob/68ec202baa7cfa9af12bcf610498fbfeba5bc299/integrations/google_ai/tests/generators/chat/test_chat_gemini.py#L258